airbyte-cdk/python/unit_tests/sources/declarative/parsers/test_model_to_component_factory.py

```diff
@@ -862,6 +863,80 @@ def test_given_data_feed_and_incremental_then_raise_error():
         )
 
 
+def test_client_side_incremental():
```

Comment:
can you add unit tests to cover the case where is_client_side_incremental is true and there is a partition router?

Reply:
added